src/subgraph/multiply.c

```diff
@@ -0,0 +1,267 @@
+    assert(values[input1_id].layout == xnn_layout_type_nhwc);
+    assert(values[input2_id].layout == xnn_layout_type_nhwc);
+    memcpy(opdata->shape1.dim, values[input1_id].shape.dim,
+           values[input1_id].shape.num_dims * sizeof(size_t));
+    memcpy(opdata->shape2.dim, values[input2_id].shape.dim,
+           values[input2_id].shape.num_dims * sizeof(size_t));
+  }
+
+  // Handle scalars. Although the output shape is dimensionless, the reshape
+  // function must be passed a valid shape to prevent skipping the op.
+  if (opdata->shape1.num_dims == 0) {
+    opdata->shape1.num_dims = 1;
+    opdata->shape1.dim[0] = 1;
+  }
+  if (opdata->shape2.num_dims == 0) {
+    opdata->shape2.num_dims = 1;
+    opdata->shape2.dim[0] = 1;
+  }
+  const size_t old_workspace_size = opdata->workspace_size;
+  enum xnn_status status = xnn_status_invalid_state;
+  switch (opdata->operator_objects[0]->type) {
+    case xnn_operator_type_multiply_nd_s32:
+      status = xnn_reshape_multiply_nd_s32(
+          opdata->operator_objects[0], opdata->shape1.num_dims,
+          opdata->shape1.dim, opdata->shape2.num_dims, opdata->shape2.dim,
+          threadpool);
+      break;
+    default:
+      XNN_UNREACHABLE;
+  }
+  if (status != xnn_status_success) {
+    return status;
+  }
+  return resize_binary_elementwise_output_tensor(
+      opdata, values, num_values, old_workspace_size, threadpool);
+}
+
+static enum xnn_status setup_multiply_operator(
+    const struct xnn_operator_data* opdata, const struct xnn_value* values,
+    size_t num_values, pthreadpool_t threadpool) {
+  const uint32_t input1_id = opdata->inputs[0];
+  assert(input1_id != XNN_INVALID_VALUE_ID);
+  assert(input1_id < num_values);
+
+  const uint32_t input2_id = opdata->inputs[1];
+  assert(input2_id != XNN_INVALID_VALUE_ID);
+  assert(input2_id < num_values);
+
+  const uint32_t output_id = opdata->outputs[0];
+  assert(output_id != XNN_INVALID_VALUE_ID);
+  assert(output_id < num_values);
+
+  const struct xnn_value* input1_value = values + input1_id;
+  const void* input1_data = input1_value->data;
+  assert(input1_data != NULL);
+
+  const struct xnn_value* input2_value = values + input2_id;
+  const void* input2_data = input2_value->data;
+  assert(input2_data != NULL);
+
+  const struct xnn_value* output_value = values + output_id;
+  void* output_data = output_value->data;
+  assert(output_data != NULL);
+
+  switch (opdata->operator_objects[0]->type) {
+    case xnn_operator_type_multiply_nd_s32:
+      return xnn_setup_multiply_nd_s32(opdata->operator_objects[0], input1_data,
+                                       input2_data, output_data);
+    default:
+      XNN_UNREACHABLE;
+  }
+}
+
+enum xnn_status xnn_define_multiply_v2(xnn_subgraph_t subgraph,
+                                       uint32_t input1_id, uint32_t input2_id,
+                                       uint32_t output_id, uint32_t flags) {
+  int32_t output_min = INT_MIN;
```

Comment:
Can this be return xnn_define_multiply(subgraph, -INFINITY, INFINITY, input1_id, input2_id, output_id, flags);

Then for s32 the create function will ignore node->output_min/max and just pass INT_MIN/INT_MAX?

Reply:
Here subgraph test is failing for s32 with xnn_define_multiply(subgraph, -INFINITY, INFINITY, input1_id, input2_id, output_id, flags); since output_min/max so,how shall we approach this?

Reply:
updated